/****************************************************************************/
// SUMO, Simulation of Urban MObility; see http://sumo.dlr.de/
// Copyright (C) 2001-2016 DLR (http://www.dlr.de/) and contributors
/****************************************************************************/
//
//   This file is part of SUMO.
//   SUMO is free software: you can redistribute it and/or modify
//   it under the terms of the GNU General Public License as published by
//   the Free Software Foundation, either version 3 of the License, or
//   (at your option) any later version.
//
/****************************************************************************/

syntax = "proto3";

package noninteracting;

option java_multiple_files = false;
option java_package = "de.dlr.sumo.noninteracting";
option java_outer_classname = "NonInteractingProto";

service PBPState {
    rpc computeWalkingTime(CMPWlkgTm) returns (PBSUMOTime) {}
    rpc getPosition(GetPos) returns (PBPosition) {}
    rpc getAngle(GetAngle) returns (PBSUMOReal) {}
    rpc getWaitingTime(PBSUMOTime) returns (PBSUMOTime) {}
    rpc getSpeed(PBMSPersonStage_Walking) returns (PBSUMOReal) {}
    rpc getNextEdge(PBMSPersonStage_Walking) returns (PBEdge) {}

}



message PBPosition {
    double x = 1;
    double y = 2;
    double z = 3;
}

message PBEdge {
    string fromJunctionId = 1;
    string toJunctionId = 2;
    double length = 3;
}

message PBSUMOTime {
    double sumoTime = 1;
}

message PBSUMOReal {
    double sumoReal = 1;
}

message PBMSPersonStage_Walking {
    PBEdge edge = 1;
    PBEdge nextRouteEdge = 2;
    double departPos = 3;
    double arrivalPos = 4;
    double maxSpeed = 5;
}

message CMPWlkgTm {
    PBEdge prev = 1;
    PBMSPersonStage_Walking stage = 2;
    PBSUMOTime sumoTime = 3;
}

message GetPos {
    PBMSPersonStage_Walking stage = 1;
    PBSUMOTime sumoTime = 2;
}

message GetAngle {
    PBMSPersonStage_Walking stage = 1;
    PBSUMOTime sumoTime = 2;
}